compiler/sizes.go

```diff
@@ -21,6 +21,11 @@ func (s *stdSizes) Alignof(T types.Type) int64 {
 	// of alignment of the elements and fields, respectively.
 	switch t := T.Underlying().(type) {
 	case *types.Array:
+		if t.Len() == 0 {
+			// 0-sized arrays, always have 0 size - and therefore can be minimally aligned
+			return s.PtrSize
```

Comment:
I think this should have an alignment of 1, not `PtrSize`. You can test this with a `Sizeof(struct{ a [0]int; b byte}{})`.

Reply:
Ya, I went back and forth on this - thank you for the clarification.

Reply:
I can update to 1 (I thought originally 1 was the correct value too).  But I was unable to corroborate that with the upstream go compiler (perhaps that is not / should not be the source of truth in this case however).

```go
package main

import (
	"unsafe"
)

type ZeroInt [0]int

type ZeroIntRaw struct {
	f [0]int
	b byte
}

type ZeroIntEmbedded struct {
	ZeroInt
	b byte
}

func main() {
	println("sizeof ZeroInt", unsafe.Sizeof(ZeroInt{}))
	println("alignof ZeroInt", unsafe.Alignof(ZeroInt{}))

	println("sizeof ZeroIntRaw", unsafe.Sizeof(ZeroIntRaw{}))
	println("alignof ZeroIntRaw", unsafe.Alignof(ZeroIntRaw{}))

	println("sizeof ZeroIntEmbedded", unsafe.Sizeof(ZeroIntEmbedded{}))
	println("alignof ZeroIntEmbedded", unsafe.Alignof(ZeroIntEmbedded{}))
}

```

The go compiler reports these all as `8`
```
sizeof ZeroInt 0
alignof ZeroInt 8
sizeof ZeroIntRaw 8
alignof ZeroIntRaw 8
sizeof ZeroIntEmbedded 8
alignof ZeroIntEmbedded 8
```

Reply:
(M1 Mac is the architecture I tried this on).